ghostwriter/static/js/tinymce/plugins/ghostwriter-comments/plugin.min.js

```diff
@@ -0,0 +1,110 @@
+tinymce.PluginManager.add('ghostwriter-comments', function(editor) {
+    // Function to decode the users hex username into ascii
+    function hex2ascii(hexString) {
+        let decodedStr = '';
+        for (let i = 0; i < hexString.length; i += 2) {
+            const byte = parseInt(hexString.substr(i, 2), 16);
+            decodedStr += String.fromCharCode(byte);
+        }
+        return decodedStr
+    }
+
+    // Create Comment
+    function createComment(comment, selection) {
+        const commentNode = editor.selection.getNode();
+        let commentThread = readComments(commentNode);
+        commentThread.push({ author: username, comment: comment });
+        editor.execCommand('mceInsertContent', false, `<span class="comment" comment-data="${encodeURIComponent(JSON.stringify(commentThread))}">${selection}</span>`);
+    }
+
+    // Read Comments
+    function readComments(commentNode) {
+        let commentThread = [];
+        if (commentNode.nodeName === 'SPAN' && commentNode.hasAttribute('comment-data')) {
+            commentDataDecoded = decodeURIComponent(commentNode.getAttribute('comment-data'));
+            commentData = JSON.parse(commentDataDecoded);
+            commentData.forEach(item => {
+                const author = item.author;
+                const comment = item.comment;
+                commentThread.push({ author: author, comment: comment });
+            });
```

Comment:
Shouldn't need to make a new array here - you should be able to just do `return commentData`.